From 8e036ae080e31965057524e7ad1d529a2fcbce02 Mon Sep 17 00:00:00 2001
From: Olivier Moysan <olivier.moysan@foss.st.com>
Date: Thu, 6 Feb 2025 10:02:43 +0100
Subject: [PATCH] arm64: dts: st: change power domain for i2s on
 stm32mp231.dtsi

The SPI/I2S peripheral supports LP-Stop1/2 low power mode.
(See "Functionalities depending on system operating mode" table in ref
manual). However, the is2 driver does not support the autonomous mode.
Change power domain to "CLUSTER_PD" for i2s nodes, on stm32mp231.dtsi.

Change-Id: I7a856d2e69fc77b82def2514239cb2986760562b
Signed-off-by: Olivier Moysan <olivier.moysan@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/434878
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
---
 arch/arm64/boot/dts/st/stm32mp231.dtsi | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/st/stm32mp231.dtsi b/arch/arm64/boot/dts/st/stm32mp231.dtsi
index 18f563a799cc..1ccb16070ccf 100644
--- a/arch/arm64/boot/dts/st/stm32mp231.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp231.dtsi
@@ -814,7 +814,7 @@ i2s2: audio-controller@400b0000 {
 					<&hpdma 52 0x43 0x21>;
 				dma-names = "rx", "tx";
 				access-controllers = <&rifsc 23>;
-				power-domains = <&RET_PD>;
+				power-domains = <&CLUSTER_PD>;
 				status = "disabled";
 			};
 
@@ -846,7 +846,7 @@ i2s3: audio-controller@400c0000 {
 					<&hpdma 54 0x43 0x21>;
 				dma-names = "rx", "tx";
 				access-controllers = <&rifsc 24>;
-				power-domains = <&RET_PD>;
+				power-domains = <&CLUSTER_PD>;
 				status = "disabled";
 			};
 
@@ -1169,7 +1169,7 @@ i2s1: audio-controller@40230000 {
 					<&hpdma 50 0x43 0x21>;
 				dma-names = "rx", "tx";
 				access-controllers = <&rifsc 22>;
-				power-domains = <&RET_PD>;
+				power-domains = <&CLUSTER_PD>;
 				status = "disabled";
 			};
 
-- 
2.50.1

